android/uhabits-android/src/main/java/org/isoron/uhabits/receivers/WidgetReceiver.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2016 Álinson Santos Xavier <isoron@gmail.com>
+ *
+ * This file is part of Loop Habit Tracker.
+ *
+ * Loop Habit Tracker is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Loop Habit Tracker is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.isoron.uhabits.receivers
+
+import android.content.*
+import android.util.*
+
+import org.isoron.uhabits.*
+import org.isoron.uhabits.core.preferences.*
+import org.isoron.uhabits.core.ui.widgets.*
+import org.isoron.uhabits.intents.*
+import org.isoron.uhabits.sync.*
+
+import dagger.*
+import org.isoron.uhabits.activities.common.dialogs.NumberPickerFactory
+import org.isoron.uhabits.core.ui.screens.habits.list.ListHabitsBehavior
+import org.isoron.uhabits.widgets.activities.NumericalCheckmarkWidgetActivity
+import android.content.Intent
+
+
+
+/**
+ * The Android BroadcastReceiver for Loop Habit Tracker.
+ *
+ *
+ * All broadcast messages are received and processed by this class.
+ */
+class WidgetReceiver : BroadcastReceiver() {
```

Comment:
Please restore the file `WidgetReceiver.java` and make your modifications there. Pull requests converting Java to Kotlin are appreciated, but please submit them separately. When you implement new features and convert languages the same time, it's really hard to spot bugs.